src/WebJobs.Script/Rpc/WebHostLanguageWorkerChannelManager.cs

```diff
@@ -102,16 +102,42 @@ public async Task SpecializeAsync()
         {
             _logger.LogInformation("Starting language worker channel specialization");
             _workerRuntime = _environment.GetEnvironmentVariable(LanguageWorkerConstants.FunctionWorkerRuntimeSettingName);
+
             ILanguageWorkerChannel languageWorkerChannel = await GetChannelAsync(_workerRuntime);
-            if (_workerRuntime != null && languageWorkerChannel != null)
+            if (languageWorkerChannel != null)
```

Comment:
change to  otherwise, shutdownIfChannelExists will throw null ref. We cannot shutdown placeholder channel here if worker runtime not set. FunctionDispatcher will take care of it.
```
public async Task SpecializeAsync()
        {
            _logger.LogInformation("Starting language worker channel specialization");
            _workerRuntime = _environment.GetEnvironmentVariable(LanguageWorkerConstants.FunctionWorkerRuntimeSettingName);

            ILanguageWorkerChannel languageWorkerChannel = await GetChannelAsync(_workerRuntime);
            if (_workerRuntime != null && languageWorkerChannel != null)
            {
                if (UsePlaceholderChannel(_workerRuntime))
                {
                    _logger.LogDebug("Loading environment variables for runtime: {runtime}", _workerRuntime);
                    await languageWorkerChannel.SendFunctionEnvironmentReloadRequest();
                }
                else
                {
                    _logger.LogDebug("Shutting down placeholder worker. Worker is not compatible for runtime: {runtime}", _workerRuntime);
                    // If we need to allow file edits, we should shutdown the webhost channel on specialization.
                    await ShutdownChannelIfExistsAsync(_workerRuntime, languageWorkerChannel.Id);
                }
            }

            _shutdownStandbyWorkerChannels();
            _logger.LogDebug("Completed language worker channel specialization");
        }

        internal bool UsePlaceholderChannel(string workerRuntime)
        {
            if (!string.IsNullOrEmpty(workerRuntime))
            {
                // Special case: node apps must be read-only to use the placeholder mode channel
                // TODO: Remove special casing when resolving https://github.com/Azure/azure-functions-host/issues/4534
                if (string.Equals(workerRuntime, LanguageWorkerConstants.NodeLanguageWorkerName))
                {
                    return _environment.FileSystemIsReadOnly();
                }
            }
            return true;
        }
```

Reply:
Ok! Should we also not make ShutdownChannelIfExists not throw? In my mind, the "IfExists" implies that this method should only try to shutdown something if it's there but otherwise continue executing safely.

Reply:
calling shutdown when workerRuntime is null is a no-op and is intentional to throw error as it enforces not calling this method with null. ShutdownChannelIfExists means that, for a valid runtime, if the list contains a channel for that runtime shut it down.